src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -692,6 +695,7 @@ export default class DefaultDeviceController
     if (!audioDevice) {
       return;
     }
+    this.muted = !enabled;
```

Comment:
is a better fix to make this one async?

Reply:
So it's actually not entirely a race condition, it's also just the mute state not being kept track of. There’s two cases where it's triggered, one is when the device is changed while it's in the middle of muting the voice focus transform. I believe this is because it creates a new transform device with the old transform device's mute state. This could be fixed through making this async or even keeping the promise and awaiting it before starting a device. The other case is when the device is set to null or an intrinsic device in this case if the device is swapped back and there was a change in the mute state before swapping, the transform will be out of sync with the mute state. This case can not be fixed by making it async.